strategies/aartur/straregy.py

```diff
@@ -0,0 +1,50 @@
+import attr
+
+from strategy import BaseMove, BaseStrategy
+from rules import State
+
+from .preparation import AdvancedState, Map
+from .solver import BaseSolver, CollectHealBonusSolver, CollectScoreBonusSolver, ShootSolver, HideSolver, CenterSolver
+
+
+@attr.s(slots=True, kw_only=True)
+class AArturBaseStrategy(BaseStrategy):
+    """My base strategy! Chicken!
+
+    Stores some Solver objects and priority coefficients associated to them.
+
+    Calculating the resulting move is quite simple:
+    1. Call all our solvers -> moves and confidences
+    2. Multiply our priority to the given confidence
+    3. Pick a move with maximum value
+    """
+
+    solvers: dict[BaseSolver, float] = attr.ib(factory=dict, init=False)
+
+    def get_next_move(self, state: State) -> BaseMove:
+        moves: list[tuple[BaseMove, float]] = []  # if only the DirectMove and Shoot were hashable...
+        state = AdvancedState(state, self.player_name)
+        mapper = Map(state)
+
+        for solver, priority in self.solvers.items():
+            move, confidence = solver.solve(state, mapper)
+            moves.append((move, priority * confidence))
```

Comment:
we can save only max value as result

Reply:
Don't need a whole list to store the moves, neither you need tuples and hashable moves at the moment.
Consider rewriting it in a streaming fashion

Reply:
I removed constructing the list from choosing the best move, but maybe i did it not in the perfect way. Check pls